projects/ng2-smart-table/src/lib/lib/grid.ts

```diff
@@ -268,4 +268,33 @@ export class Grid {
     return this.dataSet.getLastRow();
   }
 
+  private getSelectionInfo(): { perPage: number, page: number, selectedRowIndex: number, switchPageToSelectedRowPage: boolean } {
+    const switchPageToSelectedRowPage: boolean = this.getSetting('switchPageToSelectedRowPage');
+    const selectedRowIndex: number = Number(this.getSetting('selectedRowIndex', 0)) || 0;
+    const { perPage, page }: { perPage: number, page: number } = this.getSetting('pager');
+    return { perPage, page, selectedRowIndex, switchPageToSelectedRowPage };
+  }
+
+  private getRowIndexToSelect(): number {
+    const { switchPageToSelectedRowPage, selectedRowIndex, perPage } = this.getSelectionInfo();
+    const dataAmount: number = (this.source as any)?.data?.length;
+    return (
+      switchPageToSelectedRowPage &&
+      selectedRowIndex < dataAmount &&
+      selectedRowIndex >= 0
+    ) ?
+      selectedRowIndex % perPage :
+      selectedRowIndex;
+  }
+
+  private getPageToSelect(source: any): number {
+    const { switchPageToSelectedRowPage, selectedRowIndex, perPage, page } = this.getSelectionInfo();
+    let nextPage: number;
+    if (switchPageToSelectedRowPage && selectedRowIndex >= 0) {
+      nextPage = Math.ceil((selectedRowIndex + 1) / perPage) || 1;
+    }
+    const maxPageAmount: number = Math.ceil(source?.data?.length / perPage);
+    nextPage = nextPage || Math.max(1, page);
+    return maxPageAmount ? Math.min(nextPage, maxPageAmount) : nextPage;
```

Comment:
if there is no date, the function will return 1 or page in settings, if there are date then the function will return either max page or needed page.
and if there is no data it not important what page number we have because there will be rendered "no data" row

Reply:
Ok, thanks for the clarification.